internal/tiltfile/tiltfile_state.go

```diff
@@ -1478,6 +1479,28 @@ func (s *tiltfileState) translateLocal() ([]model.Manifest, error) {
 
 	return result, nil
 }
+func (s *tiltfileState) warnDuplicateYamlResources(m model.Manifest) {
+	var tempName []string
+	deployTarget := m.K8sTarget()
+	displayNames := deployTarget.DisplayNames
+
+	duplicates := make(map[string]int)
+
+	// depends on the fact that the duplicate resource has formatting of doggos:service:default:core:n where n is nth
+	//occurrence of that resource.
+	for _, name := range displayNames {
+		tempName = strings.Split(name, ":")
+		if len(tempName) > 4 { //if it is > 4 that means there's an n
+			duplicates[tempName[0]]++ // we only care about the name of the resource
+		}
+	}
+
+	for resource, numOccurences := range duplicates {
```

Comment:
(unimportant comment and you can ignore it if you've got too much to do :) ) you're only adding something to the `duplicates` map if it has more than 4 components, i.e. if it has an `n` at the end--so when you add something to that map, you already know that it appears more than once in this manifest. I believe that makes the `numOccurences > 1` check redundant. 

given that, if you wanted, you could just make `duplicates` a `map[string]bool` (this is the idiomatic go way of doing a set` and just `for entity := range duplicates { s.logger.Warnf(...) }`